lisa/platforms/platinfo.py

```diff
@@ -16,15 +16,37 @@
 #
 
 from collections.abc import Mapping
+from collections import OrderedDict
 
 from lisa.utils import HideExekallID, memoized
-from lisa.conf import DeferredValue, IntIntDict, IntListList, IntIntListDict, StrIntListDict, MultiSrcConf, KeyDesc, LevelKeyDesc, TopLevelKeyDesc
+from lisa.conf import (
+    DeferredValue, IntIntDict, IntListList, IntIntListDict, StrIntListDict,
+    MultiSrcConf, KeyDesc, LevelKeyDesc, TopLevelKeyDesc, DerivedKeyDesc
+)
 from lisa.energy_model import EnergyModel
 from lisa.wlgen.rta import RTA
 
 from devlib.target import KernelVersion
 from devlib.exception import TargetStableError
 
+def compute_capa_classes(conf):
+    """
+    Derive the platform's capacity classes from the given conf
+
+    This is intended for the creation of the ``capacity-classes`` key of
+    :class:`PlatformInfo`.
+    """
+    cpu_caps = conf['cpu-capacities']
+    capacities = sorted(set(cpu_caps.values()))
+
+    classes = [
+        [cpu for cpu, cpu_capa in cpu_caps.items() if cpu_capa == capacity]
```

Comment:
The inner list comprehension needs to be sorted as well, as it depends on the iteration order of a dictionary, which cannot be relied upon.

Also, this can be simplified using:
```
def compute_capa_classes(conf): 
     cpu_caps = conf['cpu-capacities']
     return [
         sorted(cpu_group)
         for capa, cpu_group in utils.groupby(cpu_caps.items(), key=operator.itemgetter(1))
      ]
```
`operator.itemgetter()` just returns a function that will get the `N`th item of a sequence, in this case the 2nd item of the 2-tuples returned by `dict.items()`
Could be even worth adding a `group_by_value()` helper in `utils`:
```
def group_by_value(mapping, key_sort=lambda k: 0):
     return {
        val: sorted((k for k, v in key_group), key=key_sort)
         for val, key_group in groupby(mapping.items(), key=operator.itemgetter(1))
     }
```
And then:
```
def compute_capa_classes(conf): 
     return classes = [*group_by_value(conf['cpu-capacities'], key_sort=lambda k: k).values()]
```